cmd/ddev/cmd/get.go

```diff
@@ -45,454 +24,49 @@ ddev get --remove someowner/ddev-someaddonname,
-			}
-		}
-		globalDotDdev := filepath.Join(globalconfig.GetGlobalDdevDir())
-		if len(s.GlobalFiles) > 0 {
-			util.Success("\nInstalling global components:")
-		}
-
-		globalFiles, err := fileutil.ExpandFilesAndDirectories(extractedDir, s.GlobalFiles)
-		if err != nil {
-			util.Failed("Unable to expand global files and directories: %v", err)
-		}
-		for _, file := range globalFiles {
-			src := filepath.Join(extractedDir, file)
-			dest := filepath.Join(globalDotDdev, file)
-
-			// If the file existed and had #ddev-generated OR if it did not exist, copy it in.
-			if err = fileutil.CheckSignatureOrNoFile(dest, nodeps.DdevFileSignature); err == nil {
-				err = copy.Copy(src, dest)
-				if err != nil {
-					util.Failed("Unable to copy %v to %v: %v", src, dest, err)
-				}
-				util.Success("%c %s", '\U0001F44D', file)
-			} else {
-				util.Warning("NOT overwriting %s. The #ddev-generated signature was not found in the file, so it will not be overwritten. You can remove the file and use DDEV get again if you want it to be replaced: %v", dest, err)
-			}
-		}
-		origDir, _ := os.Getwd()
-
-		defer func() {
-			err = os.Chdir(origDir)
-			if err != nil {
-				util.Failed("Unable to chdir to %v: %v", origDir, err)
-			}
-		}()
-
-		err = os.Chdir(app.GetConfigPath(""))
-		if err != nil {
-			util.Failed("Unable to chdir to %v: %v", app.GetConfigPath(""), err)
-		}
-
-		if len(s.PostInstallActions) > 0 {
-			util.Success("\nExecuting post-install actions:")
-		}
-		for i, action := range s.PostInstallActions {
-			err = ddevapp.ProcessAddonAction(action, dict, bash, verbose)
-			desc := ddevapp.GetAddonDdevDescription(action)
-			if err != nil {
-				if !verbose {
-					util.Failed("Could not process post-install action (%d) '%s'", i, desc)
-				} else {
-					util.Failed("Could not process post-install action (%d) '%s': %v", i, desc, err)
-				}
-			}
-		}
-
-		repository := ""
-		switch argType {
-		case "github":
-			repository = fmt.Sprintf("%s/%s", owner, repo)
-		case "directory":
-			fallthrough
-		case "tarball":
-			repository = sourceRepoArg
-		}
-		manifest, err := createManifestFile(app, s.Name, repository, downloadedRelease, s)
-		if err != nil {
-			util.Failed("Unable to create manifest file: %v", err)
-		}
-
-		util.Success("\nInstalled DDEV add-on %s, use `ddev restart` to enable.", sourceRepoArg)
-		if argType == "github" {
-			util.Success("Please read instructions for this add-on at the source repo at\nhttps://github.com/%v/%v\nPlease file issues and create pull requests there to improve it.", owner, repo)
-		}
-		output.UserOut.WithField("raw", manifest).Printf("Installed %s:%s from %s", manifest.Name, manifest.Version, manifest.Repository)
+		// Hand off execution for ddev get <add-on>
+		AddonGetCmd.Run(cmd, args)
```

Comment:
> Oh, does using output.stdErr automatically create a panic?

No, it's from here, we need to check the length:

```suggestion
		if len(args) > 1 {
			AddonGetCmd.Run(cmd, args)
		}
```

Reply:
Ahh, interesting. I had expected the validation of the new command to kick in and say "requires at least 1 arg(s), only received 0" but it seems that validation is skipped when running the function directly like this.

Have added back the original messaging from the old `ddev get` when no args are passed.